src/App.tsx

```diff
@@ -1,61 +1,119 @@
 import './App.scss';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './components/TodoInfo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const findUser = (userId: number) => {
+  return usersFromServer.find(user => user.id === userId) || null;
+};
+
+export const todosData: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUser(todo.userId),
+}));
 
 export const App = () => {
+  const [usertitle, setUserTitle] = useState('');
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorName, setErrorName] = useState(false);
+  const [user, setUser] = useState(0);
+  const [todos, setTodos] = useState(todosData);
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!usertitle) {
+      setErrorTitle(true);
+    }
+
+    if (!user) {
+      setErrorName(true);
+    }
+
+    if (!usertitle || !user) {
+      return;
+    }
+
+    const todoId = Math.max(...todosData.map(todo => todo.id)) + 1;
+
+    const todoItem = {
+      id: todoId,
+      title: usertitle,
+      userId: user,
+      completed: false,
+      user: findUser(user),
+    };
+
+    setTodos([...todos, todoItem]);
```

Comment:
if you need to take the previous State and modify it, you need to the callback for setTodos, please check this link: https://stackoverflow.com/questions/55823296/reactjs-prevstate-in-the-new-usestate-react-hook